### PR TITLE
Making cross compile independent of gox vender package

### PR DESCRIPTION
**Pull request by amitkrout**
## Was the change discussed in an issue?
https://github.com/openshift/odo/pull/1998#discussion_r315745452

## How to test changes?

make cross

### CODE REVIEW

**Comment by mohammedzee1000** [comment]
You can make this cleaner by doing it in a shell script and calling it here i think

**Comment by amitkrout** [comment]
```
.PHONY: cross
cross:
    @for platform in linux darwin windows ; do \
        if [ $$platform == "windows" ]; then \
        IS_EXE := .exe; \
        fi \
        go build -o dist/bin/$$platform-amd64/odo$(IS_EXE) $(BUILD_FLAGS) ./cmd/odo/ \
    done
```
I prefer this over the change i made. But unfortunately the syntax fix took longer time without success. If anybody fix the error please comment cc_ @mohammedzee1000 @kadel

File: Makefile [hunk 1]
```diff
@@ -79,10 +79,27 @@ goget-tools:
 test-coverage:
 	./scripts/generate-coverage.sh
 
+# compile for linux platform
```
Comment (mohammedzee1000) [inline Makefile]: Maybe set GOOS before doing it :
```
GOOS="darwin" go build blah
```

Comment (amitkrout) [reply Makefile]: Something like
```
.PHONY: darwin-amd64
darwin-amd64:
	GOARCH=amd64 GOOS=darwin go build $(BUILD_FLAGS) -o dist/bin/darwin-amd64/odo ./cmd/odo/
```

Comment (mohammedzee1000) [reply Makefile]: yes

File: Makefile [hunk 2]
```diff
@@ -82,7 +82,15 @@ test-coverage:
 # compile for multiple platforms
 .PHONY: cross
 cross:
-	gox -osarch="darwin/amd64 linux/amd64 windows/amd64" -output="dist/bin/{{.OS}}-{{.Arch}}/odo" $(BUILD_FLAGS) ./cmd/odo/
+	@for platform in linux darwin windows ; do \
+		if [ $$platform == "windows" ]; then \
+			echo "Cross compiling $$platform-amd64 and placing binary at dist/bin/$$platform-amd64/"; \
```
Comment (mohammedzee1000) [inline Makefile]: You can put the echo before if as its common

Comment (amitkrout) [reply Makefile]: Could not get it. Can you please explain it

**Comment by amitkrout** [comment]
> ```
> .PHONY: cross
> cross:
>     @for platform in linux darwin windows ; do \
>         if [ $$platform == "windows" ]; then \
>         IS_EXE := .exe; \
>         fi \
>         go build -o dist/bin/$$platform-amd64/odo$(IS_EXE) $(BUILD_FLAGS) ./cmd/odo/ \
>     done
> ```
> 
> I prefer this over the change i made. But unfortunately the syntax fix took longer time without success. If anybody fix the error please comment cc_ @mohammedzee1000 @kadel

Finally able to fix the syntax error.

File: Makefile (hunk 2 above)
Comment (amitkrout) [reply Makefile]: Got it, thanks
Will update it

Comment (amitkrout) [reply Makefile]: Done

**Comment by kadel** [comment]
Please remove gox from makefile and from other places if it is mentioned somewhere

**Comment by mohammedzee1000** [comment]
It works, although as stated by @kadel remove remaining references

**Comment by amitkrout** [comment]
> Please remove gox from makefile and from other places if it is mentioned somewhere

Done

**Comment by amitkrout** [comment]
```
[odo]  ✗  Failed To Update Config To Component Deployed
[odo]  ✗  Applying configuration [55s]
```
/test integration

**Comment by mohammedzee1000** [comment]
```bash
• Failure [156.587 seconds]
odoJavaE2e
/go/src/github.com/openshift/odo/tests/integration/java_test.go:13
  odo component creation
  /go/src/github.com/openshift/odo/tests/integration/java_test.go:41
    Should be able to deploy a spring boot uberjar file using openjdk [It]
    /go/src/github.com/openshift/odo/tests/integration/java_test.go:94
     Failed after 90 retries. Content in http://uberjaropenjdk-app-phppkfrfmn.apps.ci-op-dr740dig-f09f4.origin-ci-int-aws.dev.rhcloud.com doesn't include 'HTTP Booster'.
     /go/src/github.com/openshift/odo/tests/helper/helper_http.go:41
------------------------------
 Summarizing 1 Failure:
 [Fail] odoJavaE2e odo component creation [It] Should be able to deploy a spring boot uberjar file using openjdk 
/go/src/github.com/openshift/odo/tests/helper/helper_http.go:41
 Ran 4 of 97 Specs in 315.104 seconds
FAIL! -- 3 Passed | 1 Failed | 0 Pending | 93 Skipped

```
Something up with springboot

**Pull request merged**